examples/slu/speech_intent_slot/README.md

```diff
@@ -0,0 +1,134 @@
+
+### Install Dependencies
+Under the current directory, run
+```bash
+pip install -r requirements.txt
+```
+
+### Data Preparation
+1. Under the current directory, run the following script to download and process data.
+```bash
+python ../../../scripts/dataset_processing/process_slurp_data.py \
+    --data_dir="./slurp_data" \
+    --text_key="semantics" \
+    --suffix="slu"
+```
+
+2. Download evaluation code:
+```bash
+wget https://github.com/pswietojanski/slurp/raw/master/scripts/evaluation/util.py -P eval_utils/evaluation
+wget https://github.com/pswietojanski/slurp/raw/master/scripts/evaluation/metrics/distance.py -P eval_utils/evaluation/metrics
+wget https://github.com/pswietojanski/slurp/raw/master/scripts/evaluation/metrics/metrics.py -P eval_utils/evaluation/metrics
+```
+
+
+### Building Tokenizers
+1. Build the tokenizer for slu by running:
+```bash
+DATA_ROOT="./slurp_data"
+python ../../../scripts/tokenizers/process_asr_text_tokenizer.py \
+  --manifest="${DATA_ROOT}/train_slu.json,${DATA_ROOT}/train_synthetic_slu.json" \
+  --data_root="${DATA_ROOT}/tokenizers_slu/" \
+  --vocab_size=58 \
+  --tokenizer="spe" \
+  --spe_type="unigram" \
+  --log \
+  --spe_bos \
+  --spe_eos \
+  --spe_pad
+```
+
+
+### Training
+Run with the default config that uses ASR-pretrained encoder on NeMo ASR-set 3.0. The default batch size is set to 16 for a GPU with 32GB memory, please adjust it to your own case. Training for 100 epochs takes around 18 hours on a single RTX A6000 GPU with 49GB memory.
+
+```bash
+DATA_DIR="./slurp_data"
+EXP_NAME="slurp_conformer_transformer_large"
+CUDA_VISIBLE_DEVICES=0 python run_speech_intent_slot_train.py \
+    --config-path="./configs" --config-name=conformer_transformer_large_bpe \
+    model.train_ds.manifest_filepath="[${DATA_DIR}/train_slu.json,${DATA_DIR}/train_synthetic_slu.json]" \
+    model.validation_ds.manifest_filepath="${DATA_DIR}/devel_slu.json" \
+    model.test_ds.manifest_filepath="${DATA_DIR}/test_slu.json" \
+    model.tokenizer.dir="${DATA_DIR}/tokenizers_slu/tokenizer_spe_unigram_v58_pad_bos_eos" \
+    model.train_ds.batch_size=16 \
+    model.validation_ds.batch_size=16 \
+    model.test_ds.batch_size=16 \
+    trainer.devices=1 \
+    trainer.max_epochs=100 \
+    model.optim.sched.warmup_steps=2000 \
+    exp_manager.name=$EXP_NAME \
+    exp_manager.resume_if_exists=true \
+    exp_manager.resume_ignore_no_checkpoint=true
+```
+
+
+### Evaluation
+After training, we can evaluate the model by running the following script, which will first perform checkpoint averaging and then run beam search with the averaged checkpoint on the test set.
+```bash
+DATA_DIR="./slurp_data"
+EXP_NAME="slurp_conformer_transformer_large"
+CKPT_DIR="./nemo_experiments/${EXP_NAME}/checkpoints"
+CKPT_AVG_DIR="../../../examples/slu/speech_intent_slot/${CKPT_DIR}"
+
+python ../../../scripts/checkpoint_averaging/checkpoint_averaging.py $CKPT_AVG_DIR
```

Comment:
What are all these .. for? What is the "base dir" where this script will be run? Maybe copy paste it inside the example, and here keep it clean without assuming run from example dir?